clippy_lints/src/derive.rs

```diff
@@ -94,7 +94,7 @@ fn check_hash_peq<'a, 'tcx>(
     if_chain! {
         if match_path(&trait_ref.path, &paths::HASH);
         if let Some(peq_trait_def_id) = cx.tcx.lang_items().eq_trait();
-        if !&trait_ref.trait_def_id().is_local();
+       if !&trait_ref.trait_def_id().map_or(false, |def_id| def_id.is_local());
```

Comment:
formatting. But while you're at it, can you change it back to the previous version, where `if let Some(def_id) ...` and `!def_id.is_local()` was separated. I liked that better than my suggestion.

Reply:
Hm, apparently rustfmt doesn't like `if_chain!` :/

Reply:
Yep, rustfmt only works in specific/common macros